source/include/FreeRTOSIPConfigDefaults.h

```diff
@@ -252,13 +252,15 @@
 #ifdef ipconfigHAS_DEBUG_PRINTF
     #if ( ipconfigHAS_DEBUG_PRINTF == 0 )
         #ifdef FreeRTOS_debug_printf
-            #error Do not define FreeRTOS_debug_print if ipconfigHAS_DEBUG_PRINTF is set to 0
+            #warning Do not define FreeRTOS_debug_print if ipconfigHAS_DEBUG_PRINTF is set to 0
```

Comment:
The `#warning` option is not supported on windows simulator.

Since this file is not specific to STM, I think we should not make this change...